arcgis-ios-sdk-samples/Utility network/Display content of utility network container/DisplayContentUtilityNetworkContainerViewController.swift

```diff
@@ -0,0 +1,366 @@
+                    self.mapView.setViewpointCenter(point, scale: containerElement.assetType.containerViewScale) { _ in
+                        self.presentAlert(title: nil, message: "This feature contains no associations.")
+                        guard let boundingBox = self.mapView.currentViewpoint(with: .boundingGeometry)?.targetGeometry else { return }
+                        self.identifyAssociationsWithExtent(boundingBox: boundingBox)
+                    }
+                } else {
+                    boundingBox = AGSGeometryEngine.bufferGeometry(self.graphicsOverlay.extent, byDistance: 0.05)
+                    self.identifyAssociationsWithExtent(boundingBox: boundingBox!)
+                }
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+    }
+    
+    /// Get associations for the specified extent and display its associations.
+    ///
+    /// - Parameter boundingBox: The gemeotry which represents the boundaries of the extent.
+    func identifyAssociationsWithExtent(boundingBox: AGSGeometry) {
+        // Add the bounding box symbol.
+        graphicsOverlay.graphics.add(AGSGraphic(geometry: boundingBox, symbol: boundingBoxSymbol.symbol))
+        let geometry = AGSGeometryEngine.bufferGeometry(graphicsOverlay.extent, byDistance: 0.05)!
+        mapView.setViewpointGeometry(geometry) { [weak self] _ in
+            guard let self = self else { return }
+            // Get the associations for this extent to display how content features are attached or connected.
+            self.utilityNetwork?.associations(withExtent: self.graphicsOverlay.extent) { containmentAssociations, _ in
+                guard let containmentAssociations = containmentAssociations else { return }
+                containmentAssociations.forEach { association in
+                    var symbol = AGSSymbol()
+                    if association.associationType == .attachment {
+                        symbol = self.attachmentSymbol.symbol
+                    } else {
+                        symbol = self.connectivitySymbol.symbol
+                    }
+                    self.graphicsOverlay.graphics.add(AGSGraphic(geometry: association.geometry, symbol: symbol))
+                }
+                // Enable the bar button item to exit the container view.
+                self.exitBarButtonItem.isEnabled = true
+            }
+        }
+    }
+    
+    // MARK: - AGSGeoViewTouchDelegate
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Turn off user interaction to avoid straying away from the container view.
+        self.mapView.isUserInteractionEnabled = false
+        // Identify the top most feature that corresponds to the tapped point.
+        mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 5, returnPopupsOnly: false) { [weak self] (layerResults, error) in
```

Comment:
I'm not sure I understand. The doc says a completion with `[AGSIdentifyLayerResult]?` is returned.

Reply:
The completion block will give either an error, or 1 feature in the array - the **top-most feature**. Therefore, if there is no error, you may conditionally unwrap the first element in the array, and I think it is guaranteed it exists.

Reply:
I see, no problem! I think it has to stay as an array though because I have experienced instances where there are 2 values in the `layerResults` array. It occurs when 2 features are super close to one another like these 
![Screen Shot 2021-04-15 at 1 03 07 PM](https://user-images.githubusercontent.com/52215269/114931330-f4ce4900-9dea-11eb-9386-f559fddfe4ab.png)

Reply:
Ralf G. confirmed that this is expected behavior. This is what he said on Slack:

> if this is the function that you are using then you would get 2 results because it should return the top most feature from each layer